prowler/providers/gcp/lib/service/service.py

```diff
@@ -6,13 +11,61 @@ def __init__(self, service, audit_info, region="global", api_version="v1"):
         # We receive the service using __class__.__name__ or the service name in lowercase
         # e.g.: APIKeys --> we need a lowercase string, so service.lower()
         self.service = service.lower() if not service.islower() else service
-
+        self.credentials = audit_info.credentials
         self.api_version = api_version
-        self.project_ids = audit_info.project_ids
         self.default_project_id = audit_info.default_project_id
-
         self.region = region
         self.client = generate_client(service, api_version, audit_info)
+        # Only project ids that have their API enabled will be scanned
+        self.project_ids = self.__is_api_active__(audit_info.project_ids)
 
     def __get_client__(self):
         return self.client
+
+    def __location_threading_call__(self, call):
+        threads = []
+        for location in self.locations:
+            threads.append(threading.Thread(target=call, args=(location,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def __zone_threading_call__(self, call):
+        threads = []
+        for zone in self.zones:
+            threads.append(threading.Thread(target=call, args=(zone,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def __region_threading_call__(self, call):
+        threads = []
+        for region in self.regions:
+            threads.append(threading.Thread(target=call, args=(region,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
```

Comment:
I think you can have just one function for this, include a new argument with the `location/zone/region` and that's it.

Reply:
Done